p2d/p2d.py

```diff
@@ -39,16 +47,26 @@ def parse_problem():
         xml_file = f'{package_dir}/problem.xml'
         root = xml.etree.ElementTree.parse(xml_file)
         name = root.find('names').find('name').attrib['value']
-        timelimit = int(root.find('judging').find('testset').find('time-limit').text) / 1000.0
-        memlimit = int(root.find('judging').find('testset').find('memory-limit').text) // 1048576
+        judging =  root.find('judging')
+        testset = judging.find('testset')
+        timelimit = int(testset.find('time-limit').text) / 1000.0
+        memlimit = int(testset.find('memory-limit').text) // 1048576
         checker = root.find('assets').find('checker')
         interactor = root.find('assets').find('interactor')
+        input_path_pattern =testset.find('input-path-pattern').text
+        answer_path_pattern = testset.find('answer-path-pattern').text 
+        tests = []
+        for test in testset.find('tests').findall('test'):
+            method = test.attrib['method']
+            cmd = test.attrib.get('cmd', None)
```

Comment:
```xml
<test description="File &quot;test.in&quot;" method="manual"/>
```

For manual types, you can read the `description` attribute and put it into the `.desc`.